File: app/src/main/java/com/digitalsolution/waterdispenser/SplashActivity.java
package com.digitalsolution.waterdispenser;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {
    private static final int BUILD_VERSION = 29;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_activity);
        splashActivity();
    }

    private void splashActivity() {

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                int currentapiVersion = android.os.Build.VERSION.SDK_INT;
                if (BUILD_VERSION <= currentapiVersion) {
                    startActivity(new Intent(SplashActivity.this, OpenWifiNetworkAdd.class));
                } else {
                    startActivity(new Intent(SplashActivity.this, ScanQRCode.class));
                }
            }
        }, 1000);
    }
}

File: app/src/main/java/com/digitalsolution/waterdispenser/activity/SplashActivity.kt
package com.digitalsolution.waterdispenser.activity

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.digitalsolution.waterdispenser.R

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_activity)
        startActivity(Intent(this@SplashActivity, HomeActivity::class.java))
    }

    private fun splashActivity() {
        Handler().postDelayed({
            val currentapiVersion = Build.VERSION.SDK_INT
            if (BUILD_VERSION <= currentapiVersion) {
                startActivity(Intent(this@SplashActivity, OpenWifiNetworkAdd::class.java))
            } else {
                startActivity(Intent(this@SplashActivity, ScanQRCode::class.java))
            }
        }, 1000)
    }

    companion object {
        private const val BUILD_VERSION = 29
    }
}